### ISSUE TITLE
Bug: InterfaceWrongIndent assumes that spaces are used, not tabs

### ISSUE THREAD

**Issue by Daimona** [squizlabs/PHP_CodeSniffer#2836]
See bug report downstream: https://phabricator.wikimedia.org/T243529

Basically, if your project uses tabs instead of spaces, there's no way to indent interfaces properly, because PHPCS will expect 4 spaces before the interface name.

Found on 3.5.3.

**Comment by jrfnl** [comment]
@Diamona Could you share the ruleset used ? This sounds more like a configuration issue in the ruleset than a problem with the sniff. Most notably: could you check that `tabWidth` is set to `4` ?

**Comment by Daimona** [comment]
Sure: [here](https://github.com/wikimedia/mediawiki-tools-codesniffer/blob/master/MediaWiki/ruleset.xml). tabWidth isn't set, apparently.

Now, I guess that setting it would have side effects, e.g. in computing line lengths. Is it correct, and would there be other side effects? Alternatively, is there some workaround we can put in place?

**Comment by gsherwood** [comment]
Almost all PHPCS sniffs use spaces for indenting because all of the included standards require spaces over tabs. So you need to tell PHPCS how many spaces each tab represents in your project. PHPCS will then convert tabs to spaces before the code is checked.

The `Generic.WhiteSpace.DisallowSpaceIndent` sniff you are using knows about this replacement and checks the original content instead of the replaced content, so it still reports errors when spaces are used in the original code.

If you are using PHPCBF to auto-fix your code, that sniff will also replace all spaces with tabs, even when other sniffs have reformatted your code using spaces.

TLDR; set the tab width using the `--tab-width=4` CLI argument, or set it in the ruleset.xml file using `<arg name="tab-width" value="4"/>` (change `4` to whatever the project uses).

**Comment by Daimona** [comment]
Not a bug then; I'll follow-up downstream for the tab width. Thanks for your help!
